apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,301 @@
+    "AWS_REGION": os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+}
+
+
+def client_token_generator():
+    client_token = ''.join(random.choices(string.ascii_letters + string.digits, k=32))
+    return client_token
+
+"""
+Called by: create_service_by_challenge_pk (below).
+
+Registers the task definition for a new challenge, and raises a custom exception
+if the worker service is already active (which is caught by the create_service custom method.).
+"""
+
+
+def register_task_def_by_challenge_pk(client, queue_name, challenge):
+    container_name = "worker_{}".format(queue_name)
+    image = "{AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-production-worker:latest".format(AWS_ACCOUNT_ID=aws_keys["AWS_ACCOUNT_ID"])
+    AWS_DEFAULT_REGION = aws_keys["AWS_REGION"]
+    log_group = "{}_logs".format(queue_name)
+
+    task_role_arn = os.environ.get("TASK_ROLE_ARN", "")
+    execution_role_arn = os.environ.get("TASK_EXECUTION_ROLE_ARN", "")
+
+    definition = task_definition.format(queue_name=queue_name, task_role_arn=task_role_arn,
+                                        execution_role_arn=execution_role_arn, container_name=container_name,
+                                        image=image, AWS_DEFAULT_REGION=AWS_DEFAULT_REGION,
+                                        challenge_pk=challenge.pk, log_group=log_group)
+    definition = eval(definition)
+    if challenge.task_def_arn is "":
+        try:
+            response = client.register_task_definition(**definition)
+            task_def_arn = response["taskDefinition"].get("taskDefinitionArn")
+            # To store the task definition arn which will be needed for creating or updating the service later.
+            challenge.task_def_arn = task_def_arn
+            challenge.save()
+            return task_def_arn
+        except Exception as e:
+            logger.info(e)
+            raise
+    error = "Error. Task definition already registered for challenge {}.".format(challenge.pk)
+    raise Exception(error)
+
+
+# Called by the service_manager.
+def create_service_by_challenge_pk(client, challenge, client_token):
+    queue_name = challenge.queue
+    service_name = "{}_service".format(queue_name)
+    if challenge.workers is None:  # Verify if the challenge is new (i.e, service not yet created.).
+        try:
+            task_def_arn = register_task_def_by_challenge_pk(client, queue_name, challenge)
+        except Exception as e:
+            return e.response["Error"]
+
+        definition = service_definition.format(service_name=service_name, task_def_arn=task_def_arn, client_token=client_token)
+        definition = eval(definition)
+        try:
+            response = client.create_service(**definition)
+            # To verify if the service has been created on AWS.
+            if (response["ResponseMetadata"]["HTTPStatusCode"] == 200) and (response["service"]["serviceName"] == service_name):
```

Comment:
Can we use `HttpStatus` package here rather than hardcoding status codes? @KhalidRmb just check that once

Reply:
Done.